frontend/src/scenes/retention/retentionTableLogic.ts

```diff
@@ -91,6 +99,22 @@ export const retentionTableLogic = kea<retentionTableLogicType>({
         ],
         actionFilterTargetEntity: [(s) => [s.filters], (filters) => ({ events: [filters.target_entity] })],
         actionFilterReturningEntity: [(s) => [s.filters], (filters) => ({ events: [filters.returning_entity] })],
+        aggregationTargetLabel: [
+            (s) => [s.filters, s.groupTypes],
+            (
+                filters,
+                groupTypes
+            ): {
+                singular: string
+                plural: string
+            } => {
+                if (filters.aggregation_group_type_index != undefined && groupTypes.length > 0) {
```

Comment:
Suggestion: This is replicated a lot everywhere. WDYT about adding a method selector to `groupsModel` that would return this?

So something like:

```ts
    selectors: {
       aggregationLabel: [
         (s) => [s.groupTypes],
         (groupTypes) => (groupTypeIndex) => {
            if (groupTypeIndex != undefined && groupTypes.length > 0) {
              const groupType = groupTypes[groupTypeIndex]
              return { singular: groupType.group_type, plural: `${groupType.group_type}(s)` }
            }
            return { singular: 'user', plural: 'users' }
         }

```

Reply:
thanks implemented! I originally was going to do this and then went a different way and then realized i could do this again